graylog2-rest-client/src/main/java/org/graylog2/restclient/models/dashboards/widgets/FieldChartWidget.java

```diff
@@ -23,47 +23,39 @@
 
 import java.util.Map;
 
-/**
- * @author Lennart Koopmann <lennart@torch.sh>
- */
-public class FieldChartWidget extends DashboardWidget {
+public class FieldChartWidget extends ChartWidget {
 
-    private static final int DEFAULT_WIDTH = 2;
-    private static final int DEFAULT_HEIGHT = 1;
-
-    private final String streamId;
-    private final Map<String, Object> config;
+    private final String field;
+    private final String statisticalFunction;
+    private final String renderer;
+    private final String interpolation;
 
     public FieldChartWidget(Dashboard dashboard, String query, TimeRange timerange, String description, String streamId, Map<String, Object> config) {
         this(dashboard, null, description, 0, query, timerange, streamId, config, null);
     }
 
     public FieldChartWidget(Dashboard dashboard, String id, String description, int cacheTime, String query, TimeRange timerange, String streamId, Map<String, Object> config, String creatorUserId) {
-        super(Type.FIELD_CHART, id, description, cacheTime, dashboard, creatorUserId, query, timerange);
-
-        this.config = config;
+        super(Type.FIELD_CHART, id, description, cacheTime, dashboard, creatorUserId, query, timerange, streamId, (String) config.get("interval"));
 
-        if (streamId != null && !streamId.isEmpty()) {
-            this.streamId = streamId;
-        } else {
-            this.streamId = null;
-        }
+        this.field = (String) config.get("field");
+        this.statisticalFunction = (String) config.get("valuetype");
+        this.renderer = (String) config.get("renderer");
+        this.interpolation = (String) config.get("interpolation");
     }
 
     @Override
     public Map<String, Object> getConfig() {
-        Map<String, Object> c = Maps.newHashMap();
-        c.putAll(getTimerange().getQueryParams());
-        c.put("query", getQuery());
-        c.put("stream_id", streamId);
+        Map<String, Object> config = Maps.newHashMap();
+        config.putAll(getTimerange().getQueryParams());
+        config.putAll(super.getConfig());
```

Comment:
Shouldn't this be the first addition to the map so that the query parameters and the local config can overwrite settings?